commit_walker.go

```diff
@@ -6,62 +6,40 @@ import (
 	"gopkg.in/src-d/go-git.v4/core"
 )
 
-type commitWalker struct {
-	seen  map[core.Hash]bool
-	stack []*CommitIter
-	start *Commit
-	cb    func(*Commit) error
-}
-
 // WalkCommitHistory walks the commit history
 func WalkCommitHistory(c *Commit, cb func(*Commit) error) error {
```

Comment:
An awesome and very welcome simplification, thanks!

Using a queue here is quite dangerous though, as the garbage collector would not be able to free any of the already processed commits.  Consider using a stack instead, setting already processed commits' pointers to nil.